### ISSUE TITLE
Support new Ikea Tradfri

### ISSUE THREAD

**Issue by Martinvdm** [Danielhiversen/PyXiaomiGateway#142]
Is there any roadmap of supporting the new Ikea Tradfri devices through the Xiaomi Gateway? Xiaomi does support a couple of Ikea Tradfri lamps in there newest release connected to the gateway. Mi Home app is supporting those devices already.

**Comment by lonebaggie** [comment]
Unfortunately HA reports  Tradfri bulbs 

ERROR (Thread-3) [xiaomi_gateway] The device with sid xxxxxxxxxxxx isn’t supported of the used gateway firmware. Please update the gateway firmware if possible! This is the only way the issue can be solved.     

Work fine in MI Home App . Lets hope Xiaomi update the firmware to add the model name

**Comment by lonebaggie** [comment]
Same error with latest firmware  1.4.1_167.0158 shame ;(

**Comment by brendanheyu** [comment]
I have same configuration here. Six GU-10 bulbs on one switch, updated hub, bulbs are connected and controllable in Mi Home app.

**Comment by mattsaxon** [comment]
I have the same issue with an Aqara LED light, model ZNLDP12LM. According to the logs, the gateway is reporting model of '' (null). It works fine in the MI-Home app though. Might it be possible to change this component (or the wrapping HA component) so that unknown devices could be manually mapped until they are fully supported, e.g "SID [x]" = "light"

**Comment by schmurtzm** [comment]
Yes ! Any one working on this project can say if it is technically possible ?
It would be very nice to have ikea bulbs in home assistant and not only in mi-home....

**Comment by lonebaggie** [comment]
does firmware Ver. 1.4.1_170.0158 add anything ?

**Comment by JasonCampey** [comment]
Any update on this?

**Comment by bakerboy908** [comment]
About to buy a bunch of the GU10 Warm white bulbs from ikea, im planning on buying the ikea hub too, i also own the xiaomi gateway, what information would you need from these devices to get it working? 

is it hard to add the support or is it something a novice programer could attempt?

**Comment by syssi** [comment]
It's not hard if the gateway firmware supports the device. At the moment this isn't the case.

**Comment by bakerboy908** [comment]
I maybe a little confused, if i can pair the bulbs to the xiaomi hub and control them from the xiaomi app wouldnt that mean the gateway firmware supports them?

**Comment by syssi** [comment]
I'm talking about the LAN protocol which implements this library. Xiaomi extends the protocol with some delay. Your are talking about the miIO protocol (communication between gateway and MiHome app).

**Comment by dewi-ny-je** [comment]
> does firmware Ver. 1.4.1_170.0158 add anything ?

I use a Xiaomi/Aqara LED bulb tunable white and the new firmware does not solve the issue. I'm on HA 0.99.3, so thechnically this doesn't prove anything about PyXiaomiGateway, but maybe you can find out which PyXiaomiGateway version HA uses. I wasn't able to find out.

**Comment by syssi** [comment]
HA uses the most recent release of PyXaomiGateway. May be I don't understand your point.

**Comment by dewi-ny-je** [comment]
I didn't know that HA uses the latest version so I didn't know whether my information was related or not.
You asked about the changes in the firmware and I can confirm that led bulbs are still unsupported.

**Comment by robbz23** [comment]
The error comes from line 296 of /xiaomi_gateway/__init__.py.  I seems that the trådfri stuff shows up as a device that is not one of the known types that are already hard coded into the file.  I am no python expert and I have no idea how to debug the code, but I am a developer so I might be able to figure it out.  Can anybody point me in the right direction of how to enumerate devices attached to my gateway through the api?

**Comment by michaelrch** [comment]
It would be great if this functionality could be created. 
The limitation with the Tradfri gateway in HA is that you can only have one gateway. So if you want to add bulbs in place where that gateway is out of range (in an office at the end of the garden) then you can't add a second Tradfri gateway. 
But you can have as many Xiaomi gateways as you like so its a great potential solution to the Tradfi limitation. Plus by combining with the Xiaomi gateway, you could potentially skip needing a Tradfri gateway at all I guess.
